tests/back/tasks/test_tasks.py

```diff
@@ -26,36 +26,69 @@ def test_create_tabular_task():
     inputs_columns = ["SepalLengthCm", "SepalWidthCm", "PetalLengthCm", "PetalWidthCm"]
     outputs_columns = ["Species"]
-    datasetdict = to_dashai_dataset(
-        datasetdashai_csv_created, inputs_columns, outputs_columns
-    )
+    name_datasetdict = "Iris"
+    datasetdict = to_dashai_dataset(dashaidataset, inputs_columns, outputs_columns)
     tipos = {"Species": "Categorico"}
     for split in datasetdict:
         datasetdict[split] = datasetdict[split].change_columns_type(tipos)
     tabular_task = TabularClassificationTask.create()
-    tabular_task.validate_dataset_for_task(datasetdict)
+    tabular_task.validate_dataset_for_task(datasetdict, name_datasetdict)
     assert True
 
 
-def test_wrong_cardinality_task():
-    with pytest.raises(ValueError):
-        datasetdashai_csv_created = datasetdashai_from_csv("iris_extra_feature.csv")
+def test_wrong_type_task():
+    with pytest.raises(TypeError):
+        datasetdashai_csv_created = dashaidataset_from_csv("iris_extra_feature.csv")
         inputs_columns = [
             "SepalLengthCm",
             "SepalWidthCm",
             "PetalLengthCm",
             "PetalWidthCm",
         ]
         outputs_columns = ["Species", "StemCm"]
+        name_datasetdict = "Iris"
         datasetdict = to_dashai_dataset(
             datasetdashai_csv_created, inputs_columns, outputs_columns
         )
         tipos = {"Species": "Categorico"}
         for split in datasetdict:
             datasetdict[split] = datasetdict[split].change_columns_type(tipos)
         tabular_task = TabularClassificationTask.create()
-        tabular_task.validate_dataset_for_task(datasetdict)
+        tabular_task.validate_dataset_for_task(datasetdict, name_datasetdict)
+    assert True
+
+
+def test_prepare_task():
+    datasetdashai_csv_created = dashaidataset_from_csv("iris.csv")
+    inputs_columns = ["SepalLengthCm", "SepalWidthCm", "PetalLengthCm", "PetalWidthCm"]
+    outputs_columns = ["Species"]
+    name_datasetdict = "Iris"
+    datasetdict = to_dashai_dataset(
+        datasetdashai_csv_created, inputs_columns, outputs_columns
+    )
+    tabular_task = TabularClassificationTask.create()
+    datasetdict = tabular_task.prepare_for_task(datasetdict)
+    tabular_task.validate_dataset_for_task(datasetdict, name_datasetdict)
+    assert True
+
+
+def test_not_prepared_task():
+    with pytest.raises(TypeError):
+        datasetdashai_csv_created = dashaidataset_from_csv("iris.csv")
+        inputs_columns = [
+            "SepalLengthCm",
+            "SepalWidthCm",
+            "PetalLengthCm",
+            "PetalWidthCm",
+        ]
+        outputs_columns = ["Species"]
+        name_datasetdict = "Iris"
+        datasetdict = to_dashai_dataset(
+            datasetdashai_csv_created, inputs_columns, outputs_columns
+        )
+        tabular_task = TabularClassificationTask.create()
+        tabular_task.validate_dataset_for_task(datasetdict, name_datasetdict)
```

Comment:
Same comments as before:

```suggestion
def test_not_prepared_task():
    dashai_dataset_csv = dashaidataset_from_csv("iris.csv")
    inputs_columns = [
        "SepalLengthCm",
        "SepalWidthCm",
        "PetalLengthCm",
        "PetalWidthCm",
    ]
    outputs_columns = ["Species"]
    name_datasetdict = "Iris"
    
    datasetdict = to_dashai_dataset(
        dashai_dataset_csv, inputs_columns, outputs_columns
    )
    tabular_task = TabularClassificationTask.create()
    
    with pytest.raises(TypeError):
        tabular_task.validate_dataset_for_task(datasetdict, name_datasetdict)
```